modules/yugabytedb/src/main/java/org/testcontainers/containers/YugabyteDBYCQLContainer.java

```diff
@@ -0,0 +1,165 @@
+	 */
+	public YugabyteDBYCQLContainer(final String imageName) {
+		this(DockerImageName.parse(imageName));
+	}
+
+	/**
+	 * @param imageName image name
+	 */
+	public YugabyteDBYCQLContainer(final DockerImageName imageName) {
+		super(imageName);
+		imageName.assertCompatibleWith(DEFAULT_IMAGE_NAME);
+		withExposedPorts(YCQL_PORT, MASTER_DASHBOARD_PORT, TSERVER_DASHBOARD_PORT);
+		waitingFor(new YugabyteDBYCQLWaitStrategy(this).withStartupTimeout(Duration.ofSeconds(60)));
+		withCommand(ENTRYPOINT);
+	}
+
+	/**
+	 * Configures the environment variables. Setting up these variables would create the
+	 * custom objects. Setting {@link #withKeyspaceName(String)},
+	 * {@link #withUsername(String)}, {@link #withPassword(String)} these parameters will
+	 * initilaize the database with those custom values
+	 */
+	@Override
+	protected void configure() {
+		addEnv("YCQL_KEYSPACE", keyspace);
+		addEnv("YCQL_USER", username);
+		addEnv("YCQL_PASSWORD", password);
+	}
+
+	/**
+	 * @param initScript path of the initialization script file
+	 * @return {@link YugabyteDBYCQLContainer} instance
+	 */
+	public YugabyteDBYCQLContainer withInitScript(String initScript) {
+		this.initScript = initScript;
+		return this;
+	}
+
+	/**
+	 * Setting this would create the keyspace
+	 * @param keyspace keyspace
+	 * @return {@link YugabyteDBYCQLContainer} instance
+	 */
+	public YugabyteDBYCQLContainer withKeyspaceName(final String keyspace) {
+		this.keyspace = keyspace;
+		return this;
+	}
+
+	/**
+	 * Setting this would create the custom user role
+	 * @param username user name
+	 * @return {@link YugabyteDBYCQLContainer} instance
+	 */
+	public YugabyteDBYCQLContainer withUsername(final String username) {
+		this.username = username;
+		return this;
+	}
+
+	/**
+	 * Setting this along with {@link #withUsername(String)} would enable authentication
+	 * @param password password
+	 * @return {@link YugabyteDBYCQLContainer} instance
+	 */
+	public YugabyteDBYCQLContainer withPassword(final String password) {
+		this.password = password;
+		return this;
+	}
+
+	/**
+	 * Executes the initilization script
+	 * @param containerInfo containerInfo
+	 */
+	@Override
+	protected void containerIsStarted(InspectContainerResponse containerInfo) {
+		if (initScript != null) {
+			ScriptUtils.runInitScript(new YugabyteDBYCQLDelegate(getSessionBuilder()), initScript);
+		}
+	}
+
+	/**
+	 * Builds a {@link CqlSession} instance
+	 * @return {@link CqlSession} instance
+	 */
+	public CqlSession getSession() {
+		return getSessionBuilder().build();
+	}
+
+	/**
+	 * Builder method for {#com.datastax.oss.driver.api.core.CqlSession}
+	 * @return {@link CqlSessionBuilder}
+	 */
+	public CqlSessionBuilder getSessionBuilder() {
+		return CqlSession.builder().withLocalDatacenter(LOCAL_DC).withKeyspace(this.getKeyspace())
+				.withAuthCredentials(this.getUsername(), this.getPassword())
+				.addContactPoint(new InetSocketAddress(this.getHost(), this.getMappedPort(YCQL_PORT)));
+	}
```

Comment:
remove those methods, let's avoid being coupled to the client library. add 

```java
public InetSocketAddress getContactPoint() {
    return new InetSocketAddress(getHost(), getMappedPort(CQL_PORT));
}
```

See https://github.com/testcontainers/testcontainers-java/pull/2830 and the rework on cassandra container.

Reply:
I understand the dependency-related issue. We still need the client library in Delegate and WaitStrategy classes. There is a way to completely eliminate the client library, but the implementation will be a little cumbersome.